gtest/src/wasm_executor.rs

```diff
@@ -40,7 +41,8 @@ impl WasmExecutor {
         let engine = Engine::new(&config).expect("Failed to create engine");
         let mut store = Store::<StoreData<Ext>>::new(&engine, store_data);
 
-        let module = Module::new(&engine, program.code().code()).expect("Failed to create module");
+        let code = meta_binary.unwrap_or_else(|| program.code().code());
```

Comment:
program's code won't contains any other fns except init, handle, handle_reply, so any calls in gtest should be done over meta wasm only.
We can throw only meta_wasm (wout program itself) and panic if it's none

Reply:
Nice catch!